### DISCUSSION TITLE
Programmatically create pages?

### DISCUSSION THREAD

**Discussion by DavidSabine** [vuejs/vitepress#1641] [Q&A] [answered]
We're making a new Docs site with VitePress. We're loving it.

One area of the site will be a set of pages (perhaps hundreds), mostly identical, with minor variation. Imagine populating .vitepress/dist folder with something like the following:

.
├── index.html
├── stuff
│   └── index.html
├── other-stuff
│     ├── index.html
│     ├── very-interesting-stuff.html
│     └── more-interesting-things.html
└── specs
       ├── index.html
       ├── spec-1.html
       ├── spec-2…999.html
       └── spec-1000.html

The contents of the 'spec' pages are currently in JSON files. Ideally, we'd programmatically collect those .json files and use their contents to produce pages -- but do so in a way that still allows us to fully enjoy all that VitePress has to offer.

Gatsbyjs, for example, has the concept of `source` plugins, such as `gatsby-source-filesystem`. Page objects are added to the router and rendered with a template.

Open to ideas! (Anything to help save time, or point us in the right direction, or prevent us from rabbit holes.

**Comment by brc-dd** [comment, marked as answer]
Write a script to transform the data in a JSON file to a MD file and emit it to the spec directory? 👀 

There isn't a way to directly feed that data to VitePress, you'll need to generate a markdown file corresponding to each of your spec.json files and write it to filesystem before building the docs.
